File: bin/testlib.pl
#!/usr/bin/env perl

use Nullroute::Lib;

sub test_sep {
	print "\n-- @_ --\n\n";
}

sub child (&) {
	if (my $p = fork) {
		waitpid($p, 0);
	} else {
		exit shift->();
	}
}

sub foo { bar(); }
sub bar { baz(); }
sub baz { test_log(); }
sub test_log {
	_debug("debug message");
	_info("info message");
	_log("log message");
	_log2("log2 message");
	_notice("notice message");
	_warn("warning message");
	_err("error message");
	_die("fatal message");
}

test_sep("messages (normal)");

$::debug = 0; child { foo() };

test_sep("messages (debug)");

$::debug = 1; child { foo() };

test_sep("messages (debug 2)");

$::debug = 2; child { foo() };

File: bin/testlib.py
#!/usr/bin/env python3
import os
from nullroute import Core
import sys

def forked(func):
    if os.fork():
        os.wait()
    else:
        func()
        sys.exit()

def foo(): bar()

def bar(): baz()

def baz(): test_log()

def test_log():
    Core.trace("trace message")
    Core.debug("debug message")
    Core.info("info message")
    Core.notice("notice message")
    Core.warn("warning message")
    Core.err("error message")
    Core.die("fatal message", status=0)

print("\n-- messages --\n")

forked(foo)

print("\n-- messages ($DEBUG) --\n")

Core.raise_log_level(Core.LOG_TRACE)

forked(foo)
